sonar-project.properties

```diff
@@ -15,7 +15,18 @@ sonar.sourceEncoding=UTF-8
 sonar.sources=src/app
 sonar.exclusions=test-config/**, cypress/**, dist/**, .nyc_output/**, .scannerwork/**, coverage/**, **/*.module.ts, **/*.routes.ts, **/*.state.ts, **/utils.ts
 sonar.test.inclusions=**/*.spec.ts
+sonar.cpd.exclusions=src/app/forms/templates/**, src/app/api/**
 sonar.tslint.reportPaths=.reports/lint_issues.json
 sonar.javascript.lcov.reportPaths=coverage/cvs-app-vtm/lcov.info
 sonar.testExecutionReportPaths=coverage/test-report.xml
 sonar.tests=src
+
+sonar.issue.ignore.multicriteria=e1,e2,e3,e4
+sonar.issue.ignore.multicriteria.e1.ruleKey=typescript:S1128
+sonar.issue.ignore.multicriteria.e1.resourceKey=src/app/api/**
+sonar.issue.ignore.multicriteria.e2.ruleKey=typescript:S3776
+sonar.issue.ignore.multicriteria.e2.resourceKey=src/app/api/**
+sonar.issue.ignore.multicriteria.e3.ruleKey=typescript:S6325
+sonar.issue.ignore.multicriteria.e3.resourceKey=src/app/api/**
+sonar.issue.ignore.multicriteria.e4.ruleKey=typescript:S107
+sonar.issue.ignore.multicriteria.e4.resourceKey=src/app/api/**
```

Comment:
What are these ignoring :eyes:

Reply:
In order:
1. Allowing unused imports
2. Allowing cognitive complexity above 15
3. Allowing the use of `new RegExp()` instead of the `/regex/` format
4. Removing the limit to max parameters which can be passed to a function

Since these files are generated from the swagger doc, ignoring these specific rules explicitly in these files shouldn't be an issue and improves maintainability. 2 and 4 definitely are not an issue for generated files.

Reply:
No problem, makes sense to ignore. I'm wondering whether we need all the files that have been generated within the api folder as I think a lot of it is boiler plate code we haven't used. I agree the docs are useful but the code I haven't seen us use